oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -263,6 +287,69 @@ private boolean requiresFacets() {
                 .map(pd -> pd.name);
     }
 
+    private QueryBuilder similarityQuery(String mltQueryString, List<PropertyDefinition> sp) {
+        LOG.debug("parsing similarity query on {}", mltQueryString);
+        String text = null;
+        for (String param : mltQueryString.split("&")) {
+            String[] keyValuePair = param.split("=");
+            if (keyValuePair.length != 2 || keyValuePair[0] == null || keyValuePair[1] == null) {
+                throw new RuntimeException("Unparsable native query for fv similarity: " + mltQueryString);
+            } else {
+                if ("stream.body".equals(keyValuePair[0])) {
+                    text = keyValuePair[1];
+                    break;
+                }
+            }
+        }
+        BoolQueryBuilder query = boolQuery();
+        if (text != null && !sp.isEmpty()) {
+            LOG.debug("generating similarity query for {}", text);
+            String [] pathTokens = text.substring(1).split("/");
+            NodeState targetNodeState = rootState;
+            for (String token : pathTokens) {
+                targetNodeState = targetNodeState.getChildNode(token);
+            }
+            if (!targetNodeState.exists()) {
+                throw new IllegalArgumentException("Could not find node " + text);
+            }
+            for (PropertyDefinition pd : sp) {
+                int slashLastIndex = pd.name.lastIndexOf('/');
+                String propertyPath = slashLastIndex != -1 ? pd.name.substring(0, slashLastIndex) : "";
+                String propertyName = pd.name.substring(slashLastIndex+1);
+                String [] propertyPathTokens = propertyPath.split("/");
+                NodeState tempState = targetNodeState;
+                for (String token : propertyPathTokens) {
+                    if (token.isEmpty()) {
+                        break;
+                    }
+                    tempState = tempState.getChildNode(token);
+                }
+                PropertyState ps = tempState.getProperty(propertyName);
+                Blob property = ps != null ? ps.getValue(Type.BINARY) : null;
+                if (property == null) {
+                    LOG.error("Couldn't find property {} on {}", pd.name, text);
+                    continue;
+                }
+                byte[] bytes = null;
+                try {
+                    bytes = new BlobByteSource(property).read();
+                } catch (IOException e) {
+                    LOG.error("Error reading bytes from property {} on {}", pd.name, text);
```

Comment:
Here, error makes sense, because IOException would mean something wrong with the repository (e.g. Blob not found).

But the exception stack trace also needs to be logged!